modus/build.rs

```diff
@@ -14,23 +14,53 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use std::process::{Command, Stdio};
+use serde::Deserialize;
+use std::{
+    fs, io,
+    process::{Command, Stdio},
+};
 
-fn main() -> Result<(), String> {
-    let cmd = Command::new("git")
-        .stdin(Stdio::null())
-        .stderr(Stdio::inherit())
-        .stdout(Stdio::piped())
-        .args(&["rev-parse", "HEAD"])
-        .output()
-        .map_err(|e| format!("Unable to run git: {}", e))?;
-    if !cmd.status.success() {
-        return Err(format!("git rev-parse failed with {}", cmd.status));
+fn try_get_sha() -> Result<String, String> {
+    match fs::read_to_string(".cargo_vcs_info.json") {
+        Ok(vcs_json) => {
+            #[derive(Debug, Clone, Deserialize)]
+            struct VcsInfo {
+                git: VcsInfoGit,
+            }
+            #[derive(Debug, Clone, Deserialize)]
+            struct VcsInfoGit {
+                sha1: String,
+            }
+
+            let vcs_info = serde_json::from_str::<VcsInfo>(&vcs_json)
+                .map_err(|e| format!("Failed to parse .cargo_vcs_info.json: {}", e))?;
+
+            Ok(vcs_info.git.sha1)
+        }
+        Err(e) if e.kind() == io::ErrorKind::NotFound => {
+            let cmd = Command::new("git")
+                .stdin(Stdio::null())
+                .stderr(Stdio::inherit())
+                .stdout(Stdio::piped())
+                .args(&["rev-parse", "HEAD"])
+                .output()
+                .map_err(|e| format!("Unable to run git: {}", e))?;
+            if !cmd.status.success() {
+                return Err(format!("git rev-parse failed with {}", cmd.status));
+            }
```

Comment:
@maowtm 
This `git rev-parse` fails for me, maybe it's built in an directory that has access to the source code, but not the `.git`.
(If so, probably for performance reasons, `.git` might be pretty big.)

Reply:
Oh but ignore my comment ^ if it's just that the crates.io is not updated yet so it didn't try `cargo_vcs_info`.

Reply:
Should work now